cosmwasm/contracts/crosschain-registry/src/execute.rs

```diff
@@ -0,0 +1,851 @@
+use crate::helpers::*;
+use crate::state::{CHAIN_TO_CHAIN_CHANNEL_MAP, CHANNEL_ON_CHAIN_CHAIN_MAP, CONTRACT_ALIAS_MAP};
+use cosmwasm_std::{Deps, DepsMut, Response, StdError};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::ContractError;
+
+// Enum to represent the operation to be performed
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
```

Comment:
you can rename this with `#[cw_serde]` which basically translates to the same derive.  

You can also add `#[serde(untagged)]` so that the name of the enum is not used when serializing. That is, ContractAliasInput becomes: `{"contract_alias_input": {"operation": "set", ...}}` instead of `{"contract_alias_input": {"operation": {"operation": "set", ...}}}`

Reply:
I like the operation pattern a lot, btw

Reply:
Thanks! Addressed the first comment here https://github.com/osmosis-labs/osmosis/pull/4320/commits/97be4979580e1b4dc2472366e3f4c34212461f0c